src/scripts/main.js

```diff
@@ -1,87 +1,89 @@
-
-  notification.setAttribute('data-qa', 'notification');
-  notification.className = type;
-  notification.innerText = message;
-  body.insertAdjacentElement('afterbegin', notification);
-}
-
-function firstPromise() {
-  return new Promise((resolve, reject) => {
-    let isClicked = false;
-
-    const timeOut = setTimeout(() => {
-      createNotification('First promise was rejected', 'error');
-      // eslint-disable-next-line prefer-promise-reject-errors
-      reject('First promise was rejected');
-      isClicked = true;
-    }, 3000);
-
-    document.addEventListener('click', (e) => {
-      if (e.button === 0 && !isClicked) {
-        isClicked = true;
-        createNotification('First promise was resolved', 'success');
-        clearTimeout(timeOut);
-        resolve('First promise was resolved');
-      }
-    });
-  });
-}
+document.addEventListener('contextmenu', (eve) => {
+  eve.preventDefault();
+});
 
-function secondPromise() {
-  return new Promise((resolve) => {
-    let isClicked = false;
+const firstPromise = new Promise((resolve, reject) => {
+  const rejectTimeOut = setTimeout(() => {
+    const div = document.createElement('div');
 
-    document.addEventListener('mousedown', (e) => {
-      if ((e.button === 0 || e.button === 2) && !isClicked) {
-        isClicked = true;
-        createNotification('Second promise was resolved', 'success');
-        resolve('Second promise was resolved');
-      }
-    });
+    div.setAttribute('data-qa', 'notification');
+    div.classList.add('error');
+    div.textContent = 'First promise was rejected';
+    reject(div);
+  }, 3000);
+
+  document.addEventListener('click', (eve) => {
+    if (eve.button !== 0) {
+      return;
+    }
+
+    clearTimeout(rejectTimeOut);
+
+    const div = document.createElement('div');
+
+    div.setAttribute('data-qa', 'notification');
+    div.classList.add('success');
+    div.textContent = 'First promise was resolved';
+    resolve(div);
   });
-}
+});
 
-function thirdPromise() {
-  return new Promise((resolve) => {
-    let leftClick = false;
-    let rightClick = false;
+firstPromise
+  .then((div) => {
+    document.body.append(div);
+  })
+  .catch((div) => {
+    document.body.append(div);
+  });
 
-    document.addEventListener('mousedown', (e) => {
-      if (e.button === 0) {
-        leftClick = true;
-      }
+const secondPromise = new Promise((resolve) => {
+  const div = document.createElement('div');
 
-      if (e.button === 2) {
-        rightClick = true;
-      }
+  div.setAttribute('data-qa', 'notification');
 
-      if (leftClick && rightClick) {
-        createNotification('Third promise was resolved', 'success');
-        resolve('Third promise was resolved');
-        leftClick = rightClick = false;
+  const handleEvent = () => {
+    div.classList.add('success');
+    div.textContent = 'Second promise was resolved';
+    resolve(div);
+  };
+
+  document.addEventListener('contextmenu', handleEvent);
```

Comment:
The context menu event is being added multiple times, which may cause unnecessary complexity.

The click event in JavaScript provides information about which mouse button was pressed through the button property of the event object. The value of button helps differentiate between left-click (value 0), middle-click (value 1), and right-click (value 2).